### ISSUE TITLE
Package not installed when multiple package versions are pinned in dependent package

### ISSUE THREAD

**Issue by UncleGoogle** [python-poetry/poetry#6912]
- **Poetry version**: 1.2.2 (checked also on 1.1.15)
- **Python version**: 3.9.5 (checked also on 3.6.8)
- **OS version and name**: Ubuntu 20.04
- **pyproject.toml**:  https://github.com/UncleGoogle/poetry-issue-repro-wrong-lock-mark

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

when having versions pinned within a common range like that:

_pyproject.toml of dependent package_
```toml
itsdangerous = [
    {version = "2.0.1", python = "<3.10"},
    {version = "^2.0.2", python = ">=3.10"}   # originally I had a problem with ^2.0.1 so I thought it was for a common range, but apparently not
]
```

Dependency in lock have markers to higher python version only and `poetry install` on python <3.10 does not install this

_lock file_
```toml
itsdangerous = {version = "^2.0.2", markers = "python_version >= \"3.10\""}
```

### Repro
- ensure to have python 3.9
- clone https://github.com/UncleGoogle/poetry-issue-repro-wrong-lock-mark
- `poetry install`
- `poetry run python test.py`
- see ImportError because of `itsdangerous` package is not installed

**Comment by dimbleby** [comment]
I cannot reproduce.  when I run `poetry lock` in your repository, I see this diff:
```patch
 develop = true

 [package.dependencies]
-itsdangerous = {version = "^2.0.2", markers = "python_version >= \"3.10\""}
+itsdangerous = [
+    {version = "2.0.1", markers = "python_version < \"3.10\""},
+    {version = "^2.0.2", markers = "python_version >= \"3.10\""},
+]

 [package.source]
 type = "directory"
```

**Comment by UncleGoogle** [comment]
Right, I've used poetry 1.1.15 by mistake (have poetry 1.2 bin elsewhere).
So the issue is solved on 1.2, great!
